### PR TITLE
Update Python Build instructions

### PR DESCRIPTION
**Pull request by pepone**

### CODE REVIEW

File: python/BUILDING.md [hunk 1]
```diff
@@ -3,20 +3,37 @@
...
+
+```shell
+python -m ensurepip --upgrade
+```
+
+Install the build module
 
 ```shell
-pip install <URL of Ice source distribution for Python>
+pip install build
```
Comment (pepone) [inline python/BUILDING.md:25]: Should we update this to use a virtual environment and avoid issues with pip global install?

File: python/setup.py [hunk 2]
```diff
@@ -25,15 +25,7 @@
...
-packages = [
-    'Ice',
-    'IceMX',
-    'slice',
-    'slice.Glacier2',
-    'slice.Ice',
-    'slice.IceBox',
-    'slice.IceGrid',
-    'slice.IceStorm']
+packages = ['Ice', 'IceMX', 'slice']
```
Comment (pepone) [inline python/setup.py:28]: A single "slice" pseudo-package is enough to install all `.ice` files. Assuming we still want to include all the Slice files with the package.

File: python/BUILDING.md [hunk 3]
```diff
@@ -3,20 +3,37 @@
 This document describes how to build and install Ice for Python from source.
 You can also download and install a [binary distribution].
 
-* [Building with Pip](#building-with-pip)
+* [Building with Build Python Package Builder](#building-with-build-python-package-builder)
```
Comment (bernardnormier) [inline python/BUILDING.md:6]: I google "build python package builder" and there is no hit. So it's not a well-known name/title.

I would say:
Building Python Packages with build

And include a link to https://packaging.python.org/en/latest/key_projects/#build ?

File: python/BUILDING.md (hunk 1 above)
Comment (externl) [reply python/BUILDING.md:25]: I think this is a detail for users to worry about depending on their setup. There are lots of popular ways to use python like `pyenv` where using a virtual env is not required. 

If you're doing serious Python development it's very unlikely you're using the global Python.

Comment (bernardnormier) [reply python/BUILDING.md:25]: Probably.

**Review by externl — APPROVED** [review]

File: python/setup.py (hunk 2 above)
Comment (bernardnormier) [reply python/setup.py:28]: It makes sense to ship Identity.ice and other Slice files compiled in the package.

It makes less sense to ship the Slice files for IceGrid, IceStorm, Glacier2 etc.

**Review by bernardnormier — APPROVED** [review]

File: python/setup.py (hunk 2 above)
Comment (pepone) [reply python/setup.py:28]: > It makes less sense to ship the Slice files for IceGrid, IceStorm, Glacier2 etc.

How do you see us providing a Glacier2 example in this case? We could copy the required Glacier2 Slice files to the examples repository.

Comment (pepone) [reply python/setup.py:28]: Will clean-up in a follow-up PR

**Pull request merged**